README.md

```diff
@@ -1,143 +1,120 @@
-
-**Example:**
+If the consent state has already been acquired when `onConsentNotification` is called, the callback will be invoked immediately.
 
-```js
-import { onIabConsentNotification } from '@guardian/consent-management-platform';
+##### result.tcfState
 
-onIabConsentNotification((iabConsentState) => {
-	console.log(iabConsentState); // { 0: true || false || null, 1: true || false || null, ... }
-});
-```
+type: `Object` or `undefined`
 
-### CMP UI
+Reports the user's preferences to the TCFv2 purposes:
 
-The library exports The Guardian's CMP as a React component that can easily be imported into your React applications as well as a `shouldShow` function that indicates whether the user should be shown the CMP.
+```js
+{
+    1: Boolean,
+    2: Boolean,
+    // etc
+}
+```
 
-#### shouldShow
+##### result.ccpaState
 
-The `shouldShow` function returns a boolean, it will be `true` if the user does not have the appropriate consent cookies saved and `false` if they do. It takes an optional boolean `shouldRepermission`. If this is set to true it will only check for the existence of the IAB cookie, otherwise it will check for both IAB and GU_TK cookies.
+type: `Object` or `undefined`
 
-**Example:**
+Reports whether user has withdrawn consent to sell their data:
 
 ```js
-import { shouldShow } from '@guardian/consent-management-platform';
-
-shouldShow(); // true || false
+{
+    doNotSell: Boolean;
+}
 ```
 
-#### ConsentManagementPlatform React Component
-
-The properties the `ConsentManagementPlatform` component takes are listed below along with their Typescript definitions:
-
-##### onClose: () => void
+### checkUiWillShow()
 
-The `onClose` property accepts a function, this will be executed once the user has submitted their consent, either via the clicking "I'm OK with that" button in the banner, or opening the options modal selecting their choices and clicking the "Save and close" button. You can add whatever logic you want in this function. Because the `ConsentManagementPlatform` component doesn't close itself a typical example of the logic that would be included in this function might be the updating of state to hide the `ConsentManagementPlatform` component.
+returns: `Promise<Boolean>`
 
-##### source?: string
+The returned promise resolves to `true` if the user will be shown the initial privacy message, or `false` if not.
 
-The `source` property accepts an optional string. The value passed to this will be sent to the consent logs once a user has submitted their consent. The value should indicate the site on which the CMP has been seen: eg. 'manage' for 'manage.theguardian.com'. The default value passed to the logs will be 'www'.
+If it's called before `init()`, it will `reject`.
```

Comment:
does this need to reject? what if it just didn't resolve until `init` has finished? if you never call `init` then `checkUiWillShow` never resolves?

Reply:
The idea behind rejecting is that it tells the developer using this package that they've forgot to call `init` beforehand. If `init` has been called, even if the initialisation process hasn't finished yet, it resolves. Not resolving it will not make it clear to the developer that something has gone wrong with their implementation.